### PR TITLE
fix: temporary back fill of hlx config

### PR DESCRIPTION
**Pull request by solaris007**
If site-detection determines a site already exists, attempt to store the hlx config if not already present. this is only a temporary addition for back-filling the star catalogue.

### CODE REVIEW

File: src/controllers/hooks.js [hunk 1]
```diff
@@ -282,6 +299,19 @@ function HooksController(lambdaContext) {
 
     // discard the site candidate if the site exists in sites db with deliveryType=aem_edge
     if (site && site.getDeliveryType() === DELIVERY_TYPES.AEM_EDGE) {
+      // for existing site with empty hlxConfig, update it now
+      // todo: temporary to backfill hlx config for existing sites
+      /* c8 ignore start */
+      if (
+        source === SITE_CANDIDATE_SOURCES.CDN
+        && isObject(hlxConfig) && Object.keys(hlxConfig).length > 0
+        && (!isObject(site.getHlxConfig()) || Object.keys(site.getHlxConfig()).length === 0)
```
Comment (ekremney) [inline src/controllers/hooks.js:308]: idea: we can introduce a `isNonEmptyObject`method to the shared

File: src/controllers/hooks.js [hunk 2]
```diff
@@ -266,6 +266,23 @@ function buildSlackMessage(baseURL, source, hlxConfig, channel) {
 function HooksController(lambdaContext) {
   const { dataAccess } = lambdaContext;
 
+  // todo: temporary to back fill hlx config for existing sites
+  /* c8 ignore start */
```
Comment (ekremney) [inline src/controllers/hooks.js:270]: please do not introduce TODOs and `c8 ignore`s in a mature part of our codebase.

**Review by ekremney — CHANGES_REQUESTED** [review]
I like the idea in general. However, the hinting mechanism in coralogix should "ideally" work in new-value fashion. Meaning that we should not get hints for old websites. Let's see how much updates we get.

`Changes requested`: please see the comment related to TODOs and c8 ignores below

File: src/controllers/hooks.js (hunk 1 above)
Comment (solaris007) [reply src/controllers/hooks.js:308]: @ekremney here you go https://github.com/adobe/spacecat-shared/pull/280

File: src/controllers/hooks.js (hunk 2 above)
Comment (solaris007) [reply src/controllers/hooks.js:270]: i will leave the TODO as this code is temporary and needs to be removed after a while. will add test case.

**Review by ekremney — APPROVED** [review]
Looks good to me except one details. Approving to unblock, but please address the comment below:

**Pull request merged**